runtime/compiler/optimizer/JProfilingBlock.hpp

```diff
@@ -62,7 +61,7 @@ class TR_JProfilingBlock : public TR::Optimization
 
    virtual int32_t perform();
    virtual const char * optDetailString() const throw();
-
+   static TR::Node *generateBlockRawCountCalculationNode(TR::Compilation *comp, int32_t blockNumber, TR::Node *node, TR_BlockFrequencyInfo *bfi, TR_BitVector **componentCounters = NULL);
```

Comment:
This belongs on the BlockInfo not on the optimization that adds the counters - the block INFO can derive all this...

Reply:
If not on the BlockInfo then on a utility class - it is bad form to have one opt reference another which is why this would have been added.

Reply:
Changed in https://github.com/eclipse/openj9/pull/2348/commits/3296fd845e8533fd57208409e3a31aa607759f69
Now this utility function belongs to `TR_BlockFrequencyInfo`